? wxWidgets-2.7.1_wxCoord.patch
? docs/latex/.DS_Store
? docs/latex/wx/.DS_Store
? docs/latex/wx/maketex
? docs/latex/wx/tex2rtf
? docs/latex/wx/wx_acknowledgements.html
? docs/latex/wx/wx_allocatingobjects.html
? docs/latex/wx/wx_appinifunctions.html
? docs/latex/wx/wx_architecturedependency.html
? docs/latex/wx/wx_backwardcompatibility.html
? docs/latex/wx/wx_bufferclasses.html
? docs/latex/wx/wx_byteordermacros.html
? docs/latex/wx/wx_classesbycat.html
? docs/latex/wx/wx_classref.html
? docs/latex/wx/wx_clipsboard.html
? docs/latex/wx/wx_commondialogsoverview.html
? docs/latex/wx/wx_conditionalcompilation.html
? docs/latex/wx/wx_configuration.html
? docs/latex/wx/wx_constants.html
? docs/latex/wx/wx_constraintsoverview.html
? docs/latex/wx/wx_contents.html
? docs/latex/wx/wx_copyrightnotice.html
? docs/latex/wx/wx_cpp.html
? docs/latex/wx/wx_cppconst.html
? docs/latex/wx/wx_dcoverview.html
? docs/latex/wx/wx_debuggingoverview.html
? docs/latex/wx/wx_debugmacros.html
? docs/latex/wx/wx_debugstrategies.html
? docs/latex/wx/wx_dialogfunctions.html
? docs/latex/wx/wx_docviewoverview.html
? docs/latex/wx/wx_environfunctions.html
? docs/latex/wx/wx_envvars.html
? docs/latex/wx/wx_eventhandlingoverview.html
? docs/latex/wx/wx_exceptionsoverview.html
? docs/latex/wx/wx_filefunctions.html
? docs/latex/wx/wx_filehandling.html
? docs/latex/wx/wx_fs.html
? docs/latex/wx/wx_functions.html
? docs/latex/wx/wx_functionsalphabetically.html
? docs/latex/wx/wx_gdifunctions.html
? docs/latex/wx/wx_gnulicense.html
? docs/latex/wx/wx_gridoverview.html
? docs/latex/wx/wx_helloworld.html
? docs/latex/wx/wx_includefiles.html
? docs/latex/wx/wx_internationalization.html
? docs/latex/wx/wx_introduction.html
? docs/latex/wx/wx_ipcoverview.html
? docs/latex/wx/wx_keycodes.html
? docs/latex/wx/wx_keymodifiers.html
? docs/latex/wx/wx_languagecodes.html
? docs/latex/wx/wx_libraries.html
? docs/latex/wx/wx_librarieslist.html
? docs/latex/wx/wx_logfunctions.html
? docs/latex/wx/wx_makefiles.html
? docs/latex/wx/wx_mathfunctions.html
? docs/latex/wx/wx_mbconvclasses.html
? docs/latex/wx/wx_miscellany.html
? docs/latex/wx/wx_multiplat.html
? docs/latex/wx/wx_networkfunctions.html
? docs/latex/wx/wx_nonenglishoverview.html
? docs/latex/wx/wx_odbcoverview.html
? docs/latex/wx/wx_overviews.html
? docs/latex/wx/wx_platformdetails.html
? docs/latex/wx/wx_portability.html
? docs/latex/wx/wx_printersettings.html
? docs/latex/wx/wx_printingoverview.html
? docs/latex/wx/wx_processfunctions.html
? docs/latex/wx/wx_reducingerrors.html
? docs/latex/wx/wx_referencenotes.html
? docs/latex/wx/wx_requirements.html
? docs/latex/wx/wx_roughguide.html
? docs/latex/wx/wx_rttimacros.html
? docs/latex/wx/wx_runtimeclassoverview.html
? docs/latex/wx/wx_samples.html
? docs/latex/wx/wx_scrollingoverview.html
? docs/latex/wx/wx_sizeroverview.html
? docs/latex/wx/wx_stdevtid.html
? docs/latex/wx/wx_stockitems.html
? docs/latex/wx/wx_strategies.html
? docs/latex/wx/wx_stringfunctions.html
? docs/latex/wx/wx_threadfunctions.html
? docs/latex/wx/wx_timefunctions.html
? docs/latex/wx/wx_tipsoverview.html
? docs/latex/wx/wx_trefcount.html
? docs/latex/wx/wx_unicode.html
? docs/latex/wx/wx_unixprinting.html
? docs/latex/wx/wx_utilities.html
? docs/latex/wx/wx_validatoroverview.html
? docs/latex/wx/wx_versionfunctions.html
? docs/latex/wx/wx_whatis.html
? docs/latex/wx/wx_where.html
? docs/latex/wx/wx_why.html
? docs/latex/wx/wx_windowdeletionoverview.html
? docs/latex/wx/wx_windowsfiles.html
? docs/latex/wx/wx_windowstyles.html
? docs/latex/wx/wx_wxaboutdialoginfo.html
? docs/latex/wx/wx_wxacceleratorentry.html
? docs/latex/wx/wx_wxacceleratortable.html
? docs/latex/wx/wx_wxaccessible.html
? docs/latex/wx/wx_wxactivateevent.html
? docs/latex/wx/wx_wxactivexcontainer.html
? docs/latex/wx/wx_wxactivexevent.html
? docs/latex/wx/wx_wxanimation.html
? docs/latex/wx/wx_wxanimationctrl.html
? docs/latex/wx/wx_wxapp.html
? docs/latex/wx/wx_wxappoverview.html
? docs/latex/wx/wx_wxapptraits.html
? docs/latex/wx/wx_wxarc.html
? docs/latex/wx/wx_wxarchiveclassfactory.html
? docs/latex/wx/wx_wxarchiveentry.html
? docs/latex/wx/wx_wxarchiveinputstream.html
? docs/latex/wx/wx_wxarchiveiterator.html
? docs/latex/wx/wx_wxarchivenotifier.html
? docs/latex/wx/wx_wxarchiveoutputstream.html
? docs/latex/wx/wx_wxarray.html
? docs/latex/wx/wx_wxarraystring.html
? docs/latex/wx/wx_wxartprovider.html
? docs/latex/wx/wx_wxauioverview.html
? docs/latex/wx/wx_wxautobufferedpaintdc.html
? docs/latex/wx/wx_wxautomationobject.html
? docs/latex/wx/wx_wxbitmap.html
? docs/latex/wx/wx_wxbitmapbutton.html
? docs/latex/wx/wx_wxbitmapcombobox.html
? docs/latex/wx/wx_wxbitmapdataobject.html
? docs/latex/wx/wx_wxbitmaphandler.html
? docs/latex/wx/wx_wxbitmapoverview.html
? docs/latex/wx/wx_wxbookctrloverview.html
? docs/latex/wx/wx_wxboxsizer.html
? docs/latex/wx/wx_wxbrush.html
? docs/latex/wx/wx_wxbrushlist.html
? docs/latex/wx/wx_wxbuffereddc.html
? docs/latex/wx/wx_wxbufferedinputstream.html
? docs/latex/wx/wx_wxbufferedoutputstream.html
? docs/latex/wx/wx_wxbufferedpaintdc.html
? docs/latex/wx/wx_wxbusycursor.html
? docs/latex/wx/wx_wxbusyinfo.html
? docs/latex/wx/wx_wxbutton.html
? docs/latex/wx/wx_wxcalculatelayoutevent.html
? docs/latex/wx/wx_wxcalendarctrl.html
? docs/latex/wx/wx_wxcalendardateattr.html
? docs/latex/wx/wx_wxcalendarevent.html
? docs/latex/wx/wx_wxcaret.html
? docs/latex/wx/wx_wxcheckbox.html
? docs/latex/wx/wx_wxchecklistbox.html
? docs/latex/wx/wx_wxchildfocusevent.html
? docs/latex/wx/wx_wxchoice.html
? docs/latex/wx/wx_wxchoicebook.html
? docs/latex/wx/wx_wxclassinfo.html
? docs/latex/wx/wx_wxclient.html
? docs/latex/wx/wx_wxclientdata.html
? docs/latex/wx/wx_wxclientdatacontainer.html
? docs/latex/wx/wx_wxclientdc.html
? docs/latex/wx/wx_wxclipboard.html
? docs/latex/wx/wx_wxclipboardtextevent.html
? docs/latex/wx/wx_wxcloseevent.html
? docs/latex/wx/wx_wxcmdlineparser.html
? docs/latex/wx/wx_wxcollapsiblepane.html
? docs/latex/wx/wx_wxcolour.html
? docs/latex/wx/wx_wxcolourdata.html
? docs/latex/wx/wx_wxcolourdatabase.html
? docs/latex/wx/wx_wxcolourdialog.html
? docs/latex/wx/wx_wxcolourpickerctrl.html
? docs/latex/wx/wx_wxcolourpickerevent.html
? docs/latex/wx/wx_wxcombobox.html
? docs/latex/wx/wx_wxcomboctrl.html
? docs/latex/wx/wx_wxcombopopup.html
? docs/latex/wx/wx_wxcommand.html
? docs/latex/wx/wx_wxcommandevent.html
? docs/latex/wx/wx_wxcommandprocessor.html
? docs/latex/wx/wx_wxcondition.html
? docs/latex/wx/wx_wxconfigbase.html
? docs/latex/wx/wx_wxconfigoverview.html
? docs/latex/wx/wx_wxconnection.html
? docs/latex/wx/wx_wxcontaineroverview.html
? docs/latex/wx/wx_wxcontexthelp.html
? docs/latex/wx/wx_wxcontexthelpbutton.html
? docs/latex/wx/wx_wxcontextmenuevent.html
? docs/latex/wx/wx_wxcontrol.html
? docs/latex/wx/wx_wxcontrolwithitems.html
? docs/latex/wx/wx_wxcountingoutputstream.html
? docs/latex/wx/wx_wxcriticalsection.html
? docs/latex/wx/wx_wxcriticalsectionlocker.html
? docs/latex/wx/wx_wxcsconv.html
? docs/latex/wx/wx_wxcursor.html
? docs/latex/wx/wx_wxcustomdataobject.html
? docs/latex/wx/wx_wxdataformat.html
? docs/latex/wx/wx_wxdatagramsocket.html
? docs/latex/wx/wx_wxdatainputstream.html
? docs/latex/wx/wx_wxdataobject.html
? docs/latex/wx/wx_wxdataobjectcomposite.html
? docs/latex/wx/wx_wxdataobjectoverview.html
? docs/latex/wx/wx_wxdataobjectsimple.html
? docs/latex/wx/wx_wxdataoutputstream.html
? docs/latex/wx/wx_wxdataviewbitmaprenderer.html
? docs/latex/wx/wx_wxdataviewcolumn.html
? docs/latex/wx/wx_wxdataviewctrl.html
? docs/latex/wx/wx_wxdataviewcustomrenderer.html
? docs/latex/wx/wx_wxdataviewdaterenderer.html
? docs/latex/wx/wx_wxdataviewevent.html
? docs/latex/wx/wx_wxdataviewlistmodel.html
? docs/latex/wx/wx_wxdataviewlistmodelnotifier.html
? docs/latex/wx/wx_wxdataviewmodel.html
? docs/latex/wx/wx_wxdataviewprogressrenderer.html
? docs/latex/wx/wx_wxdataviewrenderer.html
? docs/latex/wx/wx_wxdataviewsortedlistmodel.html
? docs/latex/wx/wx_wxdataviewtextrenderer.html
? docs/latex/wx/wx_wxdataviewtogglerenderer.html
? docs/latex/wx/wx_wxdateevent.html
? docs/latex/wx/wx_wxdatepickerctrl.html
? docs/latex/wx/wx_wxdatespan.html
? docs/latex/wx/wx_wxdatetime.html
? docs/latex/wx/wx_wxdatetimeholidayauthority.html
? docs/latex/wx/wx_wxdatetimeoverview.html
? docs/latex/wx/wx_wxdatetimeworkdays.html
? docs/latex/wx/wx_wxdb.html
? docs/latex/wx/wx_wxdbcoldataptr.html
? docs/latex/wx/wx_wxdbcoldef.html
? docs/latex/wx/wx_wxdbcolfor.html
? docs/latex/wx/wx_wxdbcolinf.html
? docs/latex/wx/wx_wxdbconnectinf.html
? docs/latex/wx/wx_wxdbgridcolinfo.html
? docs/latex/wx/wx_wxdbgridtablebase.html
? docs/latex/wx/wx_wxdbidxdef.html
? docs/latex/wx/wx_wxdbinf.html
? docs/latex/wx/wx_wxdbtable.html
? docs/latex/wx/wx_wxdbtableinf.html
? docs/latex/wx/wx_wxdc.html
? docs/latex/wx/wx_wxdcclipper.html
? docs/latex/wx/wx_wxddeclient.html
? docs/latex/wx/wx_wxddeconnection.html
? docs/latex/wx/wx_wxddeserver.html
? docs/latex/wx/wx_wxdebugcontext.html
? docs/latex/wx/wx_wxdebugreport.html
? docs/latex/wx/wx_wxdebugreportcompress.html
? docs/latex/wx/wx_wxdebugreportpreview.html
? docs/latex/wx/wx_wxdebugreportpreviewstd.html
? docs/latex/wx/wx_wxdebugreportupload.html
? docs/latex/wx/wx_wxdebugstreambuf.html
? docs/latex/wx/wx_wxdelegaterenderernative.html
? docs/latex/wx/wx_wxdialog.html
? docs/latex/wx/wx_wxdialogoverview.html
? docs/latex/wx/wx_wxdialupevent.html
? docs/latex/wx/wx_wxdialupmanager.html
? docs/latex/wx/wx_wxdir.html
? docs/latex/wx/wx_wxdirdialog.html
? docs/latex/wx/wx_wxdirpickerctrl.html
? docs/latex/wx/wx_wxdirtraverser.html
? docs/latex/wx/wx_wxdisplay.html
? docs/latex/wx/wx_wxdllloader.html
? docs/latex/wx/wx_wxdndoverview.html
? docs/latex/wx/wx_wxdocchildframe.html
? docs/latex/wx/wx_wxdocmanager.html
? docs/latex/wx/wx_wxdocmdichildframe.html
? docs/latex/wx/wx_wxdocmdiparentframe.html
? docs/latex/wx/wx_wxdocparentframe.html
? docs/latex/wx/wx_wxdoctemplate.html
? docs/latex/wx/wx_wxdocument.html
? docs/latex/wx/wx_wxdragimage.html
? docs/latex/wx/wx_wxdropfilesevent.html
? docs/latex/wx/wx_wxdropsource.html
? docs/latex/wx/wx_wxdroptarget.html
? docs/latex/wx/wx_wxdynamiclibrary.html
? docs/latex/wx/wx_wxdynamiclibrarydetails.html
? docs/latex/wx/wx_wxencodingconverter.html
? docs/latex/wx/wx_wxeraseevent.html
? docs/latex/wx/wx_wxevent.html
? docs/latex/wx/wx_wxevthandler.html
? docs/latex/wx/wx_wxffile.html
? docs/latex/wx/wx_wxffileinputstream.html
? docs/latex/wx/wx_wxffileoutputstream.html
? docs/latex/wx/wx_wxffilestream.html
? docs/latex/wx/wx_wxfile.html
? docs/latex/wx/wx_wxfileconfig.html
? docs/latex/wx/wx_wxfiledataobject.html
? docs/latex/wx/wx_wxfiledialog.html
? docs/latex/wx/wx_wxfiledirpickerevent.html
? docs/latex/wx/wx_wxfiledroptarget.html
? docs/latex/wx/wx_wxfilehistory.html
? docs/latex/wx/wx_wxfileinputstream.html
? docs/latex/wx/wx_wxfilename.html
? docs/latex/wx/wx_wxfileoutputstream.html
? docs/latex/wx/wx_wxfileoverview.html
? docs/latex/wx/wx_wxfilepickerctrl.html
? docs/latex/wx/wx_wxfilestream.html
? docs/latex/wx/wx_wxfilesystem.html
? docs/latex/wx/wx_wxfilesystemhandler.html
? docs/latex/wx/wx_wxfiletype.html
? docs/latex/wx/wx_wxfilterinputstream.html
? docs/latex/wx/wx_wxfilteroutputstream.html
? docs/latex/wx/wx_wxfinddialogevent.html
? docs/latex/wx/wx_wxfindreplacedata.html
? docs/latex/wx/wx_wxfindreplacedialog.html
? docs/latex/wx/wx_wxflexgridsizer.html
? docs/latex/wx/wx_wxfocusevent.html
? docs/latex/wx/wx_wxfont.html
? docs/latex/wx/wx_wxfontdata.html
? docs/latex/wx/wx_wxfontdialog.html
? docs/latex/wx/wx_wxfontencodingoverview.html
? docs/latex/wx/wx_wxfontenumerator.html
? docs/latex/wx/wx_wxfontlist.html
? docs/latex/wx/wx_wxfontmapper.html
? docs/latex/wx/wx_wxfontoverview.html
? docs/latex/wx/wx_wxfontpickerctrl.html
? docs/latex/wx/wx_wxfontpickerevent.html
? docs/latex/wx/wx_wxframe.html
? docs/latex/wx/wx_wxframemanager.html
? docs/latex/wx/wx_wxfsfile.html
? docs/latex/wx/wx_wxftp.html
? docs/latex/wx/wx_wxgauge.html
? docs/latex/wx/wx_wxgbposition.html
? docs/latex/wx/wx_wxgbsizeritem.html
? docs/latex/wx/wx_wxgbspan.html
? docs/latex/wx/wx_wxgdiobject.html
? docs/latex/wx/wx_wxgenericdirctrl.html
? docs/latex/wx/wx_wxgenericvalidator.html
? docs/latex/wx/wx_wxglcanvas.html
? docs/latex/wx/wx_wxglcontext.html
? docs/latex/wx/wx_wxgrid.html
? docs/latex/wx/wx_wxgridbagsizer.html
? docs/latex/wx/wx_wxgridcellattr.html
? docs/latex/wx/wx_wxgridcellbooleditor.html
? docs/latex/wx/wx_wxgridcellboolrenderer.html
? docs/latex/wx/wx_wxgridcellchoiceeditor.html
? docs/latex/wx/wx_wxgridcelleditor.html
? docs/latex/wx/wx_wxgridcellfloateditor.html
? docs/latex/wx/wx_wxgridcellfloatrenderer.html
? docs/latex/wx/wx_wxgridcellnumbereditor.html
? docs/latex/wx/wx_wxgridcellnumberrenderer.html
? docs/latex/wx/wx_wxgridcellrenderer.html
? docs/latex/wx/wx_wxgridcellstringrenderer.html
? docs/latex/wx/wx_wxgridcelltexteditor.html
? docs/latex/wx/wx_wxgrideditorcreatedevent.html
? docs/latex/wx/wx_wxgridevent.html
? docs/latex/wx/wx_wxgridrangeselectevent.html
? docs/latex/wx/wx_wxgridsizeevent.html
? docs/latex/wx/wx_wxgridsizer.html
? docs/latex/wx/wx_wxgridtablebase.html
? docs/latex/wx/wx_wxgtkport.html
? docs/latex/wx/wx_wxhashmap.html
? docs/latex/wx/wx_wxhashset.html
? docs/latex/wx/wx_wxhashtable.html
? docs/latex/wx/wx_wxhelpcontroller.html
? docs/latex/wx/wx_wxhelpcontrollerhelpprovider.html
? docs/latex/wx/wx_wxhelpevent.html
? docs/latex/wx/wx_wxhelpprovider.html
? docs/latex/wx/wx_wxhtml.html
? docs/latex/wx/wx_wxhtmlcell.html
? docs/latex/wx/wx_wxhtmlcolourcell.html
? docs/latex/wx/wx_wxhtmlcontainercell.html
? docs/latex/wx/wx_wxhtmldcrenderer.html
? docs/latex/wx/wx_wxhtmleasyprinting.html
? docs/latex/wx/wx_wxhtmlfilter.html
? docs/latex/wx/wx_wxhtmlhelpcontroller.html
? docs/latex/wx/wx_wxhtmlhelpdata.html
? docs/latex/wx/wx_wxhtmlhelpdialog.html
? docs/latex/wx/wx_wxhtmlhelpframe.html
? docs/latex/wx/wx_wxhtmlhelpwindow.html
? docs/latex/wx/wx_wxhtmllinkinfo.html
? docs/latex/wx/wx_wxhtmllistbox.html
? docs/latex/wx/wx_wxhtmlmodalhelp.html
? docs/latex/wx/wx_wxhtmlparser.html
? docs/latex/wx/wx_wxhtmlprintout.html
? docs/latex/wx/wx_wxhtmltag.html
? docs/latex/wx/wx_wxhtmltaghandler.html
? docs/latex/wx/wx_wxhtmltagsmodule.html
? docs/latex/wx/wx_wxhtmlwidgetcell.html
? docs/latex/wx/wx_wxhtmlwindow.html
? docs/latex/wx/wx_wxhtmlwinparser.html
? docs/latex/wx/wx_wxhtmlwintaghandler.html
? docs/latex/wx/wx_wxhttp.html
? docs/latex/wx/wx_wxhyperlinkctrl.html
? docs/latex/wx/wx_wxhyperlinkevent.html
? docs/latex/wx/wx_wxicon.html
? docs/latex/wx/wx_wxiconbundle.html
? docs/latex/wx/wx_wxiconizeevent.html
? docs/latex/wx/wx_wxiconlocation.html
? docs/latex/wx/wx_wxidleevent.html
? docs/latex/wx/wx_wximage.html
? docs/latex/wx/wx_wximagehandler.html
? docs/latex/wx/wx_wximagelist.html
? docs/latex/wx/wx_wximagelistoverview.html
? docs/latex/wx/wx_wxindividuallayoutconstraint.html
? docs/latex/wx/wx_wxinitdialogevent.html
? docs/latex/wx/wx_wxinputstream.html
? docs/latex/wx/wx_wxipaddress.html
? docs/latex/wx/wx_wxipv4address.html
? docs/latex/wx/wx_wxjoystick.html
? docs/latex/wx/wx_wxjoystickevent.html
? docs/latex/wx/wx_wxkeyevent.html
? docs/latex/wx/wx_wxlayoutalgorithm.html
? docs/latex/wx/wx_wxlayoutconstraints.html
? docs/latex/wx/wx_wxlicense.html
? docs/latex/wx/wx_wxlist.html
? docs/latex/wx/wx_wxlistbook.html
? docs/latex/wx/wx_wxlistbox.html
? docs/latex/wx/wx_wxlistctrl.html
? docs/latex/wx/wx_wxlistctrloverview.html
? docs/latex/wx/wx_wxlistevent.html
? docs/latex/wx/wx_wxlistitem.html
? docs/latex/wx/wx_wxlistitemattr.html
? docs/latex/wx/wx_wxlistview.html
? docs/latex/wx/wx_wxlocale.html
? docs/latex/wx/wx_wxlog.html
? docs/latex/wx/wx_wxlogchain.html
? docs/latex/wx/wx_wxloggui.html
? docs/latex/wx/wx_wxlognull.html
? docs/latex/wx/wx_wxlogoverview.html
? docs/latex/wx/wx_wxlogpassthrough.html
? docs/latex/wx/wx_wxlogstderr.html
? docs/latex/wx/wx_wxlogstream.html
? docs/latex/wx/wx_wxlogtextctrl.html
? docs/latex/wx/wx_wxlogwindow.html
? docs/latex/wx/wx_wxlonglong.html
? docs/latex/wx/wx_wxmacport.html
? docs/latex/wx/wx_wxmask.html
? docs/latex/wx/wx_wxmaximizeevent.html
? docs/latex/wx/wx_wxmbconv.html
? docs/latex/wx/wx_wxmbconvfile.html
? docs/latex/wx/wx_wxmbconvutf16.html
? docs/latex/wx/wx_wxmbconvutf32.html
? docs/latex/wx/wx_wxmbconvutf7.html
? docs/latex/wx/wx_wxmbconvutf8.html
? docs/latex/wx/wx_wxmdichildframe.html
? docs/latex/wx/wx_wxmdiclientwindow.html
? docs/latex/wx/wx_wxmdiparentframe.html
? docs/latex/wx/wx_wxmediactrl.html
? docs/latex/wx/wx_wxmediaevent.html
? docs/latex/wx/wx_wxmeminputstream.html
? docs/latex/wx/wx_wxmemorybuffer.html
? docs/latex/wx/wx_wxmemorydc.html
? docs/latex/wx/wx_wxmemoryfshandler.html
? docs/latex/wx/wx_wxmemoutputstream.html
? docs/latex/wx/wx_wxmenu.html
? docs/latex/wx/wx_wxmenubar.html
? docs/latex/wx/wx_wxmenuevent.html
? docs/latex/wx/wx_wxmenuitem.html
? docs/latex/wx/wx_wxmessagedialog.html
? docs/latex/wx/wx_wxmetafile.html
? docs/latex/wx/wx_wxmetafiledc.html
? docs/latex/wx/wx_wxmglport.html
? docs/latex/wx/wx_wxmimetypesmanager.html
? docs/latex/wx/wx_wxminiframe.html
? docs/latex/wx/wx_wxmirrordc.html
? docs/latex/wx/wx_wxmodule.html
? docs/latex/wx/wx_wxmousecapturechangedevent.html
? docs/latex/wx/wx_wxmousecapturelostevent.html
? docs/latex/wx/wx_wxmouseevent.html
? docs/latex/wx/wx_wxmoveevent.html
? docs/latex/wx/wx_wxmswport.html
? docs/latex/wx/wx_wxmultichoicedialog.html
? docs/latex/wx/wx_wxmutex.html
? docs/latex/wx/wx_wxmutexlocker.html
? docs/latex/wx/wx_wxnode.html
? docs/latex/wx/wx_wxnotebook.html
? docs/latex/wx/wx_wxnotebookevent.html
? docs/latex/wx/wx_wxnotebooksizer.html
? docs/latex/wx/wx_wxnotifyevent.html
? docs/latex/wx/wx_wxobject.html
? docs/latex/wx/wx_wxobjectrefdata.html
? docs/latex/wx/wx_wxos2port.html
? docs/latex/wx/wx_wxoutputstream.html
? docs/latex/wx/wx_wxownerdrawncombobox.html
? docs/latex/wx/wx_wxpagesetupdialog.html
? docs/latex/wx/wx_wxpagesetupdialogdata.html
? docs/latex/wx/wx_wxpaintdc.html
? docs/latex/wx/wx_wxpaintevent.html
? docs/latex/wx/wx_wxpalette.html
? docs/latex/wx/wx_wxpalmosport.html
? docs/latex/wx/wx_wxpaneinfo.html
? docs/latex/wx/wx_wxpanel.html
? docs/latex/wx/wx_wxpasswordentrydialog.html
? docs/latex/wx/wx_wxpathlist.html
? docs/latex/wx/wx_wxpen.html
? docs/latex/wx/wx_wxpenlist.html
? docs/latex/wx/wx_wxpickerbase.html
? docs/latex/wx/wx_wxplatforminfo.html
? docs/latex/wx/wx_wxpoint.html
? docs/latex/wx/wx_wxpostscriptdc.html
? docs/latex/wx/wx_wxpowerevent.html
? docs/latex/wx/wx_wxpreviewcanvas.html
? docs/latex/wx/wx_wxpreviewcontrolbar.html
? docs/latex/wx/wx_wxpreviewframe.html
? docs/latex/wx/wx_wxprintdata.html
? docs/latex/wx/wx_wxprintdialog.html
? docs/latex/wx/wx_wxprintdialogdata.html
? docs/latex/wx/wx_wxprinter.html
? docs/latex/wx/wx_wxprinterdc.html
? docs/latex/wx/wx_wxprintout.html
? docs/latex/wx/wx_wxprintpreview.html
? docs/latex/wx/wx_wxprocess.html
? docs/latex/wx/wx_wxprocessevent.html
? docs/latex/wx/wx_wxprogressdialog.html
? docs/latex/wx/wx_wxpropertysheetdialog.html
? docs/latex/wx/wx_wxprotocol.html
? docs/latex/wx/wx_wxpython.html
? docs/latex/wx/wx_wxquantize.html
? docs/latex/wx/wx_wxquerylayoutinfoevent.html
? docs/latex/wx/wx_wxradiobox.html
? docs/latex/wx/wx_wxradiobutton.html
? docs/latex/wx/wx_wxrealpoint.html
? docs/latex/wx/wx_wxrect.html
? docs/latex/wx/wx_wxrecursionguard.html
? docs/latex/wx/wx_wxrecursionguardflag.html
? docs/latex/wx/wx_wxregex.html
? docs/latex/wx/wx_wxregion.html
? docs/latex/wx/wx_wxregioniterator.html
? docs/latex/wx/wx_wxregkey.html
? docs/latex/wx/wx_wxrenderernative.html
? docs/latex/wx/wx_wxrendererversion.html
? docs/latex/wx/wx_wxresyn.html
? docs/latex/wx/wx_wxrichtextattr.html
? docs/latex/wx/wx_wxrichtextbuffer.html
? docs/latex/wx/wx_wxrichtextcharacterstyledefinition.html
? docs/latex/wx/wx_wxrichtextctrl.html
? docs/latex/wx/wx_wxrichtextctrloverview.html
? docs/latex/wx/wx_wxrichtextevent.html
? docs/latex/wx/wx_wxrichtextfilehandler.html
? docs/latex/wx/wx_wxrichtextformattingdialog.html
? docs/latex/wx/wx_wxrichtextformattingdialogfactory.html
? docs/latex/wx/wx_wxrichtextheaderfooterdata.html
? docs/latex/wx/wx_wxrichtexthtmlhandler.html
? docs/latex/wx/wx_wxrichtextliststyledefinition.html
? docs/latex/wx/wx_wxrichtextparagraphstyledefinition.html
? docs/latex/wx/wx_wxrichtextprinting.html
? docs/latex/wx/wx_wxrichtextprintout.html
? docs/latex/wx/wx_wxrichtextrange.html
? docs/latex/wx/wx_wxrichtextstylecomboctrl.html
? docs/latex/wx/wx_wxrichtextstyledefinition.html
? docs/latex/wx/wx_wxrichtextstylelistbox.html
? docs/latex/wx/wx_wxrichtextstylesheet.html
? docs/latex/wx/wx_wxrichtextxmlhandler.html
? docs/latex/wx/wx_wxsashevent.html
? docs/latex/wx/wx_wxsashlayoutwindow.html
? docs/latex/wx/wx_wxsashwindow.html
? docs/latex/wx/wx_wxscopedarray.html
? docs/latex/wx/wx_wxscopedptr.html
? docs/latex/wx/wx_wxscopedtiedptr.html
? docs/latex/wx/wx_wxscreendc.html
? docs/latex/wx/wx_wxscrollbar.html
? docs/latex/wx/wx_wxscrolledwindow.html
? docs/latex/wx/wx_wxscrollevent.html
? docs/latex/wx/wx_wxscrollwinevent.html
? docs/latex/wx/wx_wxsemaphore.html
? docs/latex/wx/wx_wxserver.html
? docs/latex/wx/wx_wxsetcursorevent.html
? docs/latex/wx/wx_wxsimplehelpprovider.html
? docs/latex/wx/wx_wxsimplehtmllistbox.html
? docs/latex/wx/wx_wxsinglechoicedialog.html
? docs/latex/wx/wx_wxsingleinstancechecker.html
? docs/latex/wx/wx_wxsize.html
? docs/latex/wx/wx_wxsizeevent.html
? docs/latex/wx/wx_wxsizer.html
? docs/latex/wx/wx_wxsizerflags.html
? docs/latex/wx/wx_wxsizeritem.html
? docs/latex/wx/wx_wxslider.html
? docs/latex/wx/wx_wxsockaddress.html
? docs/latex/wx/wx_wxsocketbase.html
? docs/latex/wx/wx_wxsocketclient.html
? docs/latex/wx/wx_wxsocketevent.html
? docs/latex/wx/wx_wxsocketinputstream.html
? docs/latex/wx/wx_wxsocketoutputstream.html
? docs/latex/wx/wx_wxsocketserver.html
? docs/latex/wx/wx_wxsound.html
? docs/latex/wx/wx_wxspinbutton.html
? docs/latex/wx/wx_wxspinctrl.html
? docs/latex/wx/wx_wxspinevent.html
? docs/latex/wx/wx_wxsplashscreen.html
? docs/latex/wx/wx_wxsplitterevent.html
? docs/latex/wx/wx_wxsplitterrenderparams.html
? docs/latex/wx/wx_wxsplitterwindow.html
? docs/latex/wx/wx_wxsplitterwindowoverview.html
? docs/latex/wx/wx_wxstackframe.html
? docs/latex/wx/wx_wxstackwalker.html
? docs/latex/wx/wx_wxstandardpaths.html
? docs/latex/wx/wx_wxstaticbitmap.html
? docs/latex/wx/wx_wxstaticbox.html
? docs/latex/wx/wx_wxstaticboxsizer.html
? docs/latex/wx/wx_wxstaticline.html
? docs/latex/wx/wx_wxstatictext.html
? docs/latex/wx/wx_wxstatusbar.html
? docs/latex/wx/wx_wxstddialogbuttonsizer.html
? docs/latex/wx/wx_wxstopwatch.html
? docs/latex/wx/wx_wxstreambase.html
? docs/latex/wx/wx_wxstreambuffer.html
? docs/latex/wx/wx_wxstreamoverview.html
? docs/latex/wx/wx_wxstreamtotextredirector.html
? docs/latex/wx/wx_wxstring.html
? docs/latex/wx/wx_wxstringbuffer.html
? docs/latex/wx/wx_wxstringbufferlength.html
? docs/latex/wx/wx_wxstringclientdata.html
? docs/latex/wx/wx_wxstringinputstream.html
? docs/latex/wx/wx_wxstringoutputstream.html
? docs/latex/wx/wx_wxstringoverview.html
? docs/latex/wx/wx_wxstringtokenizer.html
? docs/latex/wx/wx_wxsymbolpickerdialog.html
? docs/latex/wx/wx_wxsyscolourchangedevent.html
? docs/latex/wx/wx_wxsystemoptions.html
? docs/latex/wx/wx_wxsystemsettings.html
? docs/latex/wx/wx_wxtaskbaricon.html
? docs/latex/wx/wx_wxtcpclient.html
? docs/latex/wx/wx_wxtcpconnection.html
? docs/latex/wx/wx_wxtcpserver.html
? docs/latex/wx/wx_wxtempfile.html
? docs/latex/wx/wx_wxtempfileoutputstream.html
? docs/latex/wx/wx_wxtextattr.html
? docs/latex/wx/wx_wxtextattrex.html
? docs/latex/wx/wx_wxtextctrl.html
? docs/latex/wx/wx_wxtextdataobject.html
? docs/latex/wx/wx_wxtextdroptarget.html
? docs/latex/wx/wx_wxtextentrydialog.html
? docs/latex/wx/wx_wxtextfile.html
? docs/latex/wx/wx_wxtextinputstream.html
? docs/latex/wx/wx_wxtextoutputstream.html
? docs/latex/wx/wx_wxtextvalidator.html
? docs/latex/wx/wx_wxthread.html
? docs/latex/wx/wx_wxthreadhelper.html
? docs/latex/wx/wx_wxthreadoverview.html
? docs/latex/wx/wx_wxtimer.html
? docs/latex/wx/wx_wxtimerevent.html
? docs/latex/wx/wx_wxtimespan.html
? docs/latex/wx/wx_wxtipprovider.html
? docs/latex/wx/wx_wxtipwindow.html
? docs/latex/wx/wx_wxtogglebutton.html
? docs/latex/wx/wx_wxtoolbar.html
? docs/latex/wx/wx_wxtoolbaroverview.html
? docs/latex/wx/wx_wxtoolbook.html
? docs/latex/wx/wx_wxtooltip.html
? docs/latex/wx/wx_wxtoplevelwindow.html
? docs/latex/wx/wx_wxtreebook.html
? docs/latex/wx/wx_wxtreebookevent.html
? docs/latex/wx/wx_wxtreectrl.html
? docs/latex/wx/wx_wxtreectrloverview.html
? docs/latex/wx/wx_wxtreeevent.html
? docs/latex/wx/wx_wxtreeitemdata.html
? docs/latex/wx/wx_wxupdateuievent.html
? docs/latex/wx/wx_wxuri.html
? docs/latex/wx/wx_wxurl.html
? docs/latex/wx/wx_wxurldataobject.html
? docs/latex/wx/wx_wxvalidator.html
? docs/latex/wx/wx_wxvariant.html
? docs/latex/wx/wx_wxvariantdata.html
? docs/latex/wx/wx_wxview.html
? docs/latex/wx/wx_wxvlistbox.html
? docs/latex/wx/wx_wxvscrolledwindow.html
? docs/latex/wx/wx_wxwindow.html
? docs/latex/wx/wx_wxwindowcreateevent.html
? docs/latex/wx/wx_wxwindowdc.html
? docs/latex/wx/wx_wxwindowdestroyevent.html
? docs/latex/wx/wx_wxwindowdisabler.html
? docs/latex/wx/wx_wxwindowupdatelocker.html
? docs/latex/wx/wx_wxwizard.html
? docs/latex/wx/wx_wxwizardevent.html
? docs/latex/wx/wx_wxwizardpage.html
? docs/latex/wx/wx_wxwizardpagesimple.html
? docs/latex/wx/wx_wxx11port.html
? docs/latex/wx/wx_wxxmldocument.html
? docs/latex/wx/wx_wxxmlnode.html
? docs/latex/wx/wx_wxxmlproperty.html
? docs/latex/wx/wx_wxxmlresource.html
? docs/latex/wx/wx_wxxmlresourcehandler.html
? docs/latex/wx/wx_wxzipclassfactory.html
? docs/latex/wx/wx_wxzipentry.html
? docs/latex/wx/wx_wxzipinputstream.html
? docs/latex/wx/wx_wxzipnotifier.html
? docs/latex/wx/wx_wxzipoutputstream.html
? docs/latex/wx/wx_wxzlibinputstream.html
? docs/latex/wx/wx_wxzliboutputstream.html
? docs/latex/wx/wx_xrcoverview.html
? include/wx/.DS_Store
? include/wx/mac/setup.h
Index: docs/latex/wx/function.tex
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/docs/latex/wx/function.tex,v
retrieving revision 1.261
diff -b -u -2 -r1.261 function.tex
--- docs/latex/wx/function.tex	2006/10/25 00:26:55	1.261
+++ docs/latex/wx/function.tex	2006/10/29 01:45:07
@@ -59,4 +59,5 @@
 \helpref{wxConcatFiles}{wxconcatfiles}\\
 \helpref{wxConstCast}{wxconstcast}\\
+\helpref{wxCoordRound}{wxcoordround}\\
 \helpref{wxCopyFile}{wxcopyfile}\\
 \helpref{wxCreateDynamicObject}{wxcreatedynamicobject}\\
@@ -2845,4 +2846,19 @@
 code which might have to be compiled with an old compiler without support for
 this language feature but still take advantage of it when it is available.
+
+
+
+\membersection{::wxCoordRound}\label{wxcoordround}
+
+\func{wxCoord}{wxCoordRound}{\param{const float\& }{f}}
+
+\func{wxCoord}{wxCoordRound}{\param{const double\& }{f}}
+
+Convert \em{f} to a wxCoord, using round-to-nearest. This is commonly used
+in scaling calculations.
+
+\wxheading{Include files}
+
+<wx/defs.h>
 
 
Index: include/wx/defs.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/defs.h,v
retrieving revision 1.559
diff -b -u -2 -r1.559 defs.h
--- include/wx/defs.h	2006/10/20 09:16:41	1.559
+++ include/wx/defs.h	2006/10/29 01:45:08
@@ -659,4 +659,16 @@
 enum {  wxDefaultCoord = -1 };
 
+/*  round-to-nearest (used in scaling) */
+#ifdef __cplusplus
+inline wxCoord wxCoordRound(const float& f)
+{
+    return (f > 0) ? (wxCoord)(f + 0.5) : (f < 0) ? (wxCoord)(f - 0.5) : 0;
+}
+inline wxCoord wxCoordRound(const double& f)
+{
+    return (f > 0) ? (wxCoord)(f + 0.5) : (f < 0) ? (wxCoord)(f - 0.5) : 0;
+}
+#endif
+
 /*  ---------------------------------------------------------------------------- */
 /*  define fixed length types */
Index: include/wx/dfb/dc.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/dfb/dc.h,v
retrieving revision 1.5
diff -b -u -2 -r1.5 dc.h
--- include/wx/dfb/dc.h	2006/10/24 12:26:27	1.5
+++ include/wx/dfb/dc.h	2006/10/29 01:45:08
@@ -81,61 +81,33 @@
     wxCoord XDEV2LOG(wxCoord x) const
     {
-        wxCoord new_x = x - m_deviceOriginX;
-        if (new_x > 0)
-            return (wxCoord)((double)(new_x) / m_scaleX + 0.5) * m_signX + m_logicalOriginX;
-        else
-            return (wxCoord)((double)(new_x) / m_scaleX - 0.5) * m_signX + m_logicalOriginX;
+        return wxCoordRound((double)(x - m_deviceOriginX) / m_scaleX) * m_signX + m_logicalOriginX;
     }
     wxCoord XDEV2LOGREL(wxCoord x) const
     {
-        if (x > 0)
-            return (wxCoord)((double)(x) / m_scaleX + 0.5);
-        else
-            return (wxCoord)((double)(x) / m_scaleX - 0.5);
+        return wxCoordRound((double)(x) / m_scaleX);
     }
     wxCoord YDEV2LOG(wxCoord y) const
     {
-        wxCoord new_y = y - m_deviceOriginY;
-        if (new_y > 0)
-            return (wxCoord)((double)(new_y) / m_scaleY + 0.5) * m_signY + m_logicalOriginY;
-        else
-            return (wxCoord)((double)(new_y) / m_scaleY - 0.5) * m_signY + m_logicalOriginY;
+        return wxCoordRound((double)(y - m_deviceOriginY) / m_scaleY) * m_signY + m_logicalOriginY;
     }
     wxCoord YDEV2LOGREL(wxCoord y) const
     {
-        if (y > 0)
-            return (wxCoord)((double)(y) / m_scaleY + 0.5);
-        else
-            return (wxCoord)((double)(y) / m_scaleY - 0.5);
+        return wCoordRound((double)(y) / m_scaleY);
     }
     wxCoord XLOG2DEV(wxCoord x) const
     {
-        wxCoord new_x = x - m_logicalOriginX;
-        if (new_x > 0)
-            return (wxCoord)((double)(new_x) * m_scaleX + 0.5) * m_signX + m_deviceOriginX;
-        else
-            return (wxCoord)((double)(new_x) * m_scaleX - 0.5) * m_signX + m_deviceOriginX;
+        return wxCoordRound((double)(x - m_logicalOriginX) * m_scaleX) * m_signX + m_deviceOriginX;
     }
     wxCoord XLOG2DEVREL(wxCoord x) const
     {
-        if (x > 0)
-            return (wxCoord)((double)(x) * m_scaleX + 0.5);
-        else
-            return (wxCoord)((double)(x) * m_scaleX - 0.5);
+        return wxCoordRound((double)(x) * m_scaleX);
     }
     wxCoord YLOG2DEV(wxCoord y) const
     {
-        wxCoord new_y = y - m_logicalOriginY;
-        if (new_y > 0)
-            return (wxCoord)((double)(new_y) * m_scaleY + 0.5) * m_signY + m_deviceOriginY;
-        else
-            return (wxCoord)((double)(new_y) * m_scaleY - 0.5) * m_signY + m_deviceOriginY;
+        return wxCoordRound((double)(y - m_logicalOriginY) * m_scaleY) * m_signY + m_deviceOriginY;
     }
     wxCoord YLOG2DEVREL(wxCoord y) const
     {
-        if (y > 0)
-            return (wxCoord)((double)(y) * m_scaleY + 0.5);
-        else
-            return (wxCoord)((double)(y) * m_scaleY - 0.5);
+        return wxCoordRound((double)(y) * m_scaleY);
     }
 
Index: include/wx/gtk/gnome/gprint.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/gtk/gnome/gprint.h,v
retrieving revision 1.16
diff -b -u -2 -r1.16 gprint.h
--- include/wx/gtk/gnome/gprint.h	2006/10/24 16:32:01	1.16
+++ include/wx/gtk/gnome/gprint.h	2006/10/29 01:45:08
@@ -297,61 +297,33 @@
     wxCoord XDEV2LOG(wxCoord x) const
     {
-        wxCoord new_x = x - m_deviceOriginX;
-        if (new_x > 0)
-            return (wxCoord)((double)(new_x) / m_scaleX + 0.5) * m_signX + m_logicalOriginX;
-        else
-            return (wxCoord)((double)(new_x) / m_scaleX - 0.5) * m_signX + m_logicalOriginX;
+        return wxCoordRound((double)(x - m_deviceOriginX) / m_scaleX) * m_signX + m_logicalOriginX;
     }
     wxCoord XDEV2LOGREL(wxCoord x) const
     {
-        if (x > 0)
-            return (wxCoord)((double)(x) / m_scaleX + 0.5);
-        else
-            return (wxCoord)((double)(x) / m_scaleX - 0.5);
+        return wxCoordRound((double)(x) / m_scaleX);
     }
     wxCoord YDEV2LOG(wxCoord y) const
     {
-        wxCoord new_y = y - m_deviceOriginY;
-        if (new_y > 0)
-            return (wxCoord)((double)(new_y) / m_scaleY + 0.5) * m_signY + m_logicalOriginY;
-        else
-            return (wxCoord)((double)(new_y) / m_scaleY - 0.5) * m_signY + m_logicalOriginY;
+        return wxCoordRound((double)(y - m_deviceOriginY) / m_scaleY) * m_signY + m_logicalOriginY;
     }
     wxCoord YDEV2LOGREL(wxCoord y) const
     {
-        if (y > 0)
-            return (wxCoord)((double)(y) / m_scaleY + 0.5);
-        else
-            return (wxCoord)((double)(y) / m_scaleY - 0.5);
+        return wCoordRound((double)(y) / m_scaleY);
     }
     wxCoord XLOG2DEV(wxCoord x) const
     {
-        wxCoord new_x = x - m_logicalOriginX;
-        if (new_x > 0)
-            return (wxCoord)((double)(new_x) * m_scaleX + 0.5) * m_signX + m_deviceOriginX;
-        else
-            return (wxCoord)((double)(new_x) * m_scaleX - 0.5) * m_signX + m_deviceOriginX;
+        return wxCoordRound((double)(x - m_logicalOriginX) * m_scaleX) * m_signX + m_deviceOriginX;
     }
     wxCoord XLOG2DEVREL(wxCoord x) const
     {
-        if (x > 0)
-            return (wxCoord)((double)(x) * m_scaleX + 0.5);
-        else
-            return (wxCoord)((double)(x) * m_scaleX - 0.5);
+        return wxCoordRound((double)(x) * m_scaleX);
     }
     wxCoord YLOG2DEV(wxCoord y) const
     {
-        wxCoord new_y = y - m_logicalOriginY;
-        if (new_y > 0)
-            return (wxCoord)((double)(new_y) * m_scaleY + 0.5) * m_signY + m_deviceOriginY;
-        else
-            return (wxCoord)((double)(new_y) * m_scaleY - 0.5) * m_signY + m_deviceOriginY;
+        return wxCoordRound((double)(y - m_logicalOriginY) * m_scaleY) * m_signY + m_deviceOriginY;
     }
     wxCoord YLOG2DEVREL(wxCoord y) const
     {
-        if (y > 0)
-            return (wxCoord)((double)(y) * m_scaleY + 0.5);
-        else
-            return (wxCoord)((double)(y) * m_scaleY - 0.5);
+        return wxCoordRound((double)(y) * m_scaleY);
     }
 private:
Index: include/wx/gtk1/dc.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/gtk1/dc.h,v
retrieving revision 1.35
diff -b -u -2 -r1.35 dc.h
--- include/wx/gtk1/dc.h	2006/10/09 16:39:02	1.35
+++ include/wx/gtk1/dc.h	2006/10/29 01:45:08
@@ -71,61 +71,33 @@
     wxCoord XDEV2LOG(wxCoord x) const
     {
-        wxCoord new_x = x - m_deviceOriginX;
-        if (new_x > 0)
-            return (wxCoord)((double)(new_x) / m_scaleX + 0.5) * m_signX + m_logicalOriginX;
-        else
-            return (wxCoord)((double)(new_x) / m_scaleX - 0.5) * m_signX + m_logicalOriginX;
+        return wxCoordRound((double)(x - m_deviceOriginX) / m_scaleX) * m_signX + m_logicalOriginX;
     }
     wxCoord XDEV2LOGREL(wxCoord x) const
     {
-        if (x > 0)
-            return (wxCoord)((double)(x) / m_scaleX + 0.5);
-        else
-            return (wxCoord)((double)(x) / m_scaleX - 0.5);
+        return wxCoordRound((double)(x) / m_scaleX);
     }
     wxCoord YDEV2LOG(wxCoord y) const
     {
-        wxCoord new_y = y - m_deviceOriginY;
-        if (new_y > 0)
-            return (wxCoord)((double)(new_y) / m_scaleY + 0.5) * m_signY + m_logicalOriginY;
-        else
-            return (wxCoord)((double)(new_y) / m_scaleY - 0.5) * m_signY + m_logicalOriginY;
+        return wxCoordRound((double)(y - m_deviceOriginY) / m_scaleY) * m_signY + m_logicalOriginY;
     }
     wxCoord YDEV2LOGREL(wxCoord y) const
     {
-        if (y > 0)
-            return (wxCoord)((double)(y) / m_scaleY + 0.5);
-        else
-            return (wxCoord)((double)(y) / m_scaleY - 0.5);
+        return wCoordRound((double)(y) / m_scaleY);
     }
     wxCoord XLOG2DEV(wxCoord x) const
     {
-        wxCoord new_x = x - m_logicalOriginX;
-        if (new_x > 0)
-            return (wxCoord)((double)(new_x) * m_scaleX + 0.5) * m_signX + m_deviceOriginX;
-        else
-            return (wxCoord)((double)(new_x) * m_scaleX - 0.5) * m_signX + m_deviceOriginX;
+        return wxCoordRound((double)(x - m_logicalOriginX) * m_scaleX) * m_signX + m_deviceOriginX;
     }
     wxCoord XLOG2DEVREL(wxCoord x) const
     {
-        if (x > 0)
-            return (wxCoord)((double)(x) * m_scaleX + 0.5);
-        else
-            return (wxCoord)((double)(x) * m_scaleX - 0.5);
+        return wxCoordRound((double)(x) * m_scaleX);
     }
     wxCoord YLOG2DEV(wxCoord y) const
     {
-        wxCoord new_y = y - m_logicalOriginY;
-        if (new_y > 0)
-            return (wxCoord)((double)(new_y) * m_scaleY + 0.5) * m_signY + m_deviceOriginY;
-        else
-            return (wxCoord)((double)(new_y) * m_scaleY - 0.5) * m_signY + m_deviceOriginY;
+        return wxCoordRound((double)(y - m_logicalOriginY) * m_scaleY) * m_signY + m_deviceOriginY;
     }
     wxCoord YLOG2DEVREL(wxCoord y) const
     {
-        if (y > 0)
-            return (wxCoord)((double)(y) * m_scaleY + 0.5);
-        else
-            return (wxCoord)((double)(y) * m_scaleY - 0.5);
+        return wxCoordRound((double)(y) * m_scaleY);
     }
 
Index: include/wx/mac/carbon/dc.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/mac/carbon/dc.h,v
retrieving revision 1.24
diff -b -u -2 -r1.24 dc.h
--- include/wx/mac/carbon/dc.h	2006/10/21 17:46:33	1.24
+++ include/wx/mac/carbon/dc.h	2006/10/29 01:45:08
@@ -159,86 +159,43 @@
     wxCoord XDEV2LOG(wxCoord x) const
     {
-        long new_x = x - m_deviceOriginX;
-        if (new_x > 0)
-            return (wxCoord)((double)new_x / m_scaleX + 0.5) * m_signX + m_logicalOriginX;
-        else
-            return (wxCoord)((double)new_x / m_scaleX - 0.5) * m_signX + m_logicalOriginX;
+        return wxCoordRound((double)(x - m_deviceOriginX) / m_scaleX) * m_signX + m_logicalOriginX;
     }
-
     wxCoord XDEV2LOGREL(wxCoord x) const
     {
-        if (x > 0)
-            return (wxCoord)((double)x / m_scaleX + 0.5);
-        else
-            return (wxCoord)((double)x / m_scaleX - 0.5);
+        return wxCoordRound((double)(x) / m_scaleX);
     }
-
     wxCoord YDEV2LOG(wxCoord y) const
     {
-        long new_y = y - m_deviceOriginY;
-        if (new_y > 0)
-            return (wxCoord)((double)new_y / m_scaleY + 0.5) * m_signY + m_logicalOriginY;
-        else
-            return (wxCoord)((double)new_y / m_scaleY - 0.5) * m_signY + m_logicalOriginY;
+        return wxCoordRound((double)(y - m_deviceOriginY) / m_scaleY) * m_signY + m_logicalOriginY;
     }
-
     wxCoord YDEV2LOGREL(wxCoord y) const
     {
-        if (y > 0)
-            return (wxCoord)((double)y / m_scaleY + 0.5);
-        else
-            return (wxCoord)((double)y / m_scaleY - 0.5);
+        return wCoordRound((double)(y) / m_scaleY);
     }
-
     wxCoord XLOG2DEV(wxCoord x) const
     {
-        long new_x = x - m_logicalOriginX;
-        if (new_x > 0)
-            return (wxCoord)((double)new_x * m_scaleX + 0.5) * m_signX + m_deviceOriginX;
-        else
-            return (wxCoord)((double)new_x * m_scaleX - 0.5) * m_signX + m_deviceOriginX;
+        return wxCoordRound((double)(x - m_logicalOriginX) * m_scaleX) * m_signX + m_deviceOriginX;
     }
-
     wxCoord XLOG2DEVREL(wxCoord x) const
     {
-        if (x > 0)
-            return (wxCoord)((double)x * m_scaleX + 0.5);
-        else
-            return (wxCoord)((double)x * m_scaleX - 0.5);
+        return wxCoordRound((double)(x) * m_scaleX);
     }
-
     wxCoord YLOG2DEV(wxCoord y) const
     {
-        long new_y = y - m_logicalOriginY;
-        if (new_y > 0)
-            return (wxCoord)((double)new_y * m_scaleY + 0.5) * m_signY + m_deviceOriginY;
-        else
-            return (wxCoord)((double)new_y * m_scaleY - 0.5) * m_signY + m_deviceOriginY;
+        return wxCoordRound((double)(y - m_logicalOriginY) * m_scaleY) * m_signY + m_deviceOriginY;
     }
-
     wxCoord YLOG2DEVREL(wxCoord y) const
     {
-        if (y > 0)
-            return (wxCoord)((double)y * m_scaleY + 0.5);
-        else
-            return (wxCoord)((double)y * m_scaleY - 0.5);
+        return wxCoordRound((double)(y) * m_scaleY);
     }
 
     wxCoord XLOG2DEVMAC(wxCoord x) const
     {
-        long new_x = x - m_logicalOriginX;
-        if (new_x > 0)
-            return (wxCoord)((double)new_x * m_scaleX + 0.5) * m_signX + m_deviceOriginX + m_macLocalOrigin.x;
-        else
-            return (wxCoord)((double)new_x * m_scaleX - 0.5) * m_signX + m_deviceOriginX + m_macLocalOrigin.x;
+        return wxCoordRound((double)(x - m_logicalOriginX) * m_scaleX) * m_signX + m_deviceOriginX + m_macLocalOrigin.x;
     }
 
     wxCoord YLOG2DEVMAC(wxCoord y) const
     {
-        long new_y = y - m_logicalOriginY;
-        if (new_y > 0)
-            return (wxCoord)((double)new_y * m_scaleY + 0.5) * m_signY + m_deviceOriginY + m_macLocalOrigin.y;
-        else
-            return (wxCoord)((double)new_y * m_scaleY - 0.5) * m_signY + m_deviceOriginY + m_macLocalOrigin.y;
+        return wxCoordRound((double)(y - m_logicalOriginY) * m_scaleY) * m_signY + m_deviceOriginY + m_macLocalOrigin.y;
     }
 
Index: include/wx/mac/classic/dc.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/mac/classic/dc.h,v
retrieving revision 1.8
diff -b -u -2 -r1.8 dc.h
--- include/wx/mac/classic/dc.h	2006/09/05 20:45:58	1.8
+++ include/wx/mac/classic/dc.h	2006/10/29 01:45:08
@@ -106,77 +106,43 @@
     wxCoord XDEV2LOG(wxCoord x) const
     {
-      long new_x = x - m_deviceOriginX ;
-      if (new_x > 0)
-        return (wxCoord)((double)(new_x) / m_scaleX + 0.5) * m_signX + m_logicalOriginX;
-      else
-        return (wxCoord)((double)(new_x) / m_scaleX - 0.5) * m_signX + m_logicalOriginX;
+        return wxCoordRound((double)(x - m_deviceOriginX) / m_scaleX) * m_signX + m_logicalOriginX;
     }
     wxCoord XDEV2LOGREL(wxCoord x) const
     {
-      if (x > 0)
-        return (wxCoord)((double)(x) / m_scaleX + 0.5);
-      else
-        return (wxCoord)((double)(x) / m_scaleX - 0.5);
+        return wxCoordRound((double)(x) / m_scaleX);
     }
     wxCoord YDEV2LOG(wxCoord y) const
     {
-      long new_y = y - m_deviceOriginY ;
-      if (new_y > 0)
-        return (wxCoord)((double)(new_y) / m_scaleY + 0.5) * m_signY + m_logicalOriginY;
-      else
-        return (wxCoord)((double)(new_y) / m_scaleY - 0.5) * m_signY + m_logicalOriginY;
+        return wxCoordRound((double)(y - m_deviceOriginY) / m_scaleY) * m_signY + m_logicalOriginY;
     }
     wxCoord YDEV2LOGREL(wxCoord y) const
     {
-      if (y > 0)
-        return (wxCoord)((double)(y) / m_scaleY + 0.5);
-      else
-        return (wxCoord)((double)(y) / m_scaleY - 0.5);
+        return wCoordRound((double)(y) / m_scaleY);
     }
     wxCoord XLOG2DEV(wxCoord x) const
     {
-      long new_x = x - m_logicalOriginX;
-      if (new_x > 0)
-        return (wxCoord)((double)(new_x) * m_scaleX + 0.5) * m_signX + m_deviceOriginX  ;
-      else
-        return (wxCoord)((double)(new_x) * m_scaleX - 0.5) * m_signX + m_deviceOriginX ;
+        return wxCoordRound((double)(x - m_logicalOriginX) * m_scaleX) * m_signX + m_deviceOriginX;
     }
     wxCoord XLOG2DEVREL(wxCoord x) const
     {
-      if (x > 0)
-        return (wxCoord)((double)(x) * m_scaleX + 0.5);
-      else
-        return (wxCoord)((double)(x) * m_scaleX - 0.5);
+        return wxCoordRound((double)(x) * m_scaleX);
     }
     wxCoord YLOG2DEV(wxCoord y) const
     {
-      long new_y = y - m_logicalOriginY ;
-      if (new_y > 0)
-        return (wxCoord)((double)(new_y) * m_scaleY + 0.5) * m_signY + m_deviceOriginY ;
-      else
-        return (wxCoord)((double)(new_y) * m_scaleY - 0.5) * m_signY + m_deviceOriginY ;
+        return wxCoordRound((double)(y - m_logicalOriginY) * m_scaleY) * m_signY + m_deviceOriginY;
     }
     wxCoord YLOG2DEVREL(wxCoord y) const
     {
-      if (y > 0)
-        return (wxCoord)((double)(y) * m_scaleY + 0.5);
-      else
-        return (wxCoord)((double)(y) * m_scaleY - 0.5);
+        return wxCoordRound((double)(y) * m_scaleY);
     }
+
     wxCoord XLOG2DEVMAC(wxCoord x) const
     {
-      long new_x = x - m_logicalOriginX;
-      if (new_x > 0)
-        return (wxCoord)((double)(new_x) * m_scaleX + 0.5) * m_signX + m_deviceOriginX + m_macLocalOrigin.x ;
-      else
-        return (wxCoord)((double)(new_x) * m_scaleX - 0.5) * m_signX + m_deviceOriginX + m_macLocalOrigin.x ;
+        return wxCoordRound((double)(x - m_logicalOriginX) * m_scaleX) * m_signX + m_deviceOriginX + m_macLocalOrigin.x;
     }
+
     wxCoord YLOG2DEVMAC(wxCoord y) const
     {
-      long new_y = y - m_logicalOriginY ;
-      if (new_y > 0)
-        return (wxCoord)((double)(new_y) * m_scaleY + 0.5) * m_signY + m_deviceOriginY + m_macLocalOrigin.y ;
-      else
-        return (wxCoord)((double)(new_y) * m_scaleY - 0.5) * m_signY + m_deviceOriginY + m_macLocalOrigin.y ;
+        return wxCoordRound((double)(y - m_logicalOriginY) * m_scaleY) * m_signY + m_deviceOriginY + m_macLocalOrigin.y;
     }
 
Index: include/wx/mgl/dc.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/mgl/dc.h,v
retrieving revision 1.17
diff -b -u -2 -r1.17 dc.h
--- include/wx/mgl/dc.h	2006/09/05 20:46:07	1.17
+++ include/wx/mgl/dc.h	2006/10/29 01:45:09
@@ -102,61 +102,33 @@
     wxCoord XDEV2LOG(wxCoord x) const
     {
-        wxCoord new_x = x - m_deviceOriginX;
-        if (new_x > 0)
-            return (wxCoord)((double)(new_x) / m_scaleX + 0.5) * m_signX + m_logicalOriginX;
-        else
-            return (wxCoord)((double)(new_x) / m_scaleX - 0.5) * m_signX + m_logicalOriginX;
+        return wxCoordRound((double)(x - m_deviceOriginX) / m_scaleX) * m_signX + m_logicalOriginX;
     }
     wxCoord XDEV2LOGREL(wxCoord x) const
     {
-        if (x > 0)
-            return (wxCoord)((double)(x) / m_scaleX + 0.5);
-        else
-            return (wxCoord)((double)(x) / m_scaleX - 0.5);
+        return wxCoordRound((double)(x) / m_scaleX);
     }
     wxCoord YDEV2LOG(wxCoord y) const
     {
-        wxCoord new_y = y - m_deviceOriginY;
-        if (new_y > 0)
-            return (wxCoord)((double)(new_y) / m_scaleY + 0.5) * m_signY + m_logicalOriginY;
-        else
-            return (wxCoord)((double)(new_y) / m_scaleY - 0.5) * m_signY + m_logicalOriginY;
+        return wxCoordRound((double)(y - m_deviceOriginY) / m_scaleY) * m_signY + m_logicalOriginY;
     }
     wxCoord YDEV2LOGREL(wxCoord y) const
     {
-        if (y > 0)
-            return (wxCoord)((double)(y) / m_scaleY + 0.5);
-        else
-            return (wxCoord)((double)(y) / m_scaleY - 0.5);
+        return wCoordRound((double)(y) / m_scaleY);
     }
     wxCoord XLOG2DEV(wxCoord x) const
     {
-        wxCoord new_x = x - m_logicalOriginX;
-        if (new_x > 0)
-            return (wxCoord)((double)(new_x) * m_scaleX + 0.5) * m_signX + m_deviceOriginX;
-        else
-            return (wxCoord)((double)(new_x) * m_scaleX - 0.5) * m_signX + m_deviceOriginX;
+        return wxCoordRound((double)(x - m_logicalOriginX) * m_scaleX) * m_signX + m_deviceOriginX;
     }
     wxCoord XLOG2DEVREL(wxCoord x) const
     {
-        if (x > 0)
-            return (wxCoord)((double)(x) * m_scaleX + 0.5);
-        else
-            return (wxCoord)((double)(x) * m_scaleX - 0.5);
+        return wxCoordRound((double)(x) * m_scaleX);
     }
     wxCoord YLOG2DEV(wxCoord y) const
     {
-        wxCoord new_y = y - m_logicalOriginY;
-        if (new_y > 0)
-            return (wxCoord)((double)(new_y) * m_scaleY + 0.5) * m_signY + m_deviceOriginY;
-        else
-            return (wxCoord)((double)(new_y) * m_scaleY - 0.5) * m_signY + m_deviceOriginY;
+        return wxCoordRound((double)(y - m_logicalOriginY) * m_scaleY) * m_signY + m_deviceOriginY;
     }
     wxCoord YLOG2DEVREL(wxCoord y) const
     {
-        if (y > 0)
-            return (wxCoord)((double)(y) * m_scaleY + 0.5);
-        else
-            return (wxCoord)((double)(y) * m_scaleY - 0.5);
+        return wxCoordRound((double)(y) * m_scaleY);
     }
 
Index: include/wx/motif/dc.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/motif/dc.h,v
retrieving revision 1.23
diff -b -u -2 -r1.23 dc.h
--- include/wx/motif/dc.h	2006/09/05 20:46:11	1.23
+++ include/wx/motif/dc.h	2006/10/29 01:45:09
@@ -73,79 +73,43 @@
     wxCoord XDEV2LOG(wxCoord x) const
     {
-        wxCoord new_x = x - m_deviceOriginX;
-        if (new_x > 0)
-            return (wxCoord)((double)(new_x) / m_scaleX + 0.5) * m_signX + m_logicalOriginX;
-        else
-            return (wxCoord)((double)(new_x) / m_scaleX - 0.5) * m_signX + m_logicalOriginX;
+        return wxCoordRound((double)(x - m_deviceOriginX) / m_scaleX) * m_signX + m_logicalOriginX;
     }
     wxCoord XDEV2LOGREL(wxCoord x) const
     {
-        if (x > 0)
-            return (wxCoord)((double)(x) / m_scaleX + 0.5);
-        else
-            return (wxCoord)((double)(x) / m_scaleX - 0.5);
+        return wxCoordRound((double)(x) / m_scaleX);
     }
     wxCoord YDEV2LOG(wxCoord y) const
     {
-        wxCoord new_y = y - m_deviceOriginY;
-        if (new_y > 0)
-            return (wxCoord)((double)(new_y) / m_scaleY + 0.5) * m_signY + m_logicalOriginY;
-        else
-            return (wxCoord)((double)(new_y) / m_scaleY - 0.5) * m_signY + m_logicalOriginY;
+        return wxCoordRound((double)(y - m_deviceOriginY) / m_scaleY) * m_signY + m_logicalOriginY;
     }
     wxCoord YDEV2LOGREL(wxCoord y) const
     {
-        if (y > 0)
-            return (wxCoord)((double)(y) / m_scaleY + 0.5);
-        else
-            return (wxCoord)((double)(y) / m_scaleY - 0.5);
+        return wCoordRound((double)(y) / m_scaleY);
     }
     wxCoord XLOG2DEV(wxCoord x) const
     {
-        wxCoord new_x = x - m_logicalOriginX;
-        if (new_x > 0)
-            return (wxCoord)((double)(new_x) * m_scaleX + 0.5) * m_signX + m_deviceOriginX;
-        else
-            return (wxCoord)((double)(new_x) * m_scaleX - 0.5) * m_signX + m_deviceOriginX;
+        return wxCoordRound((double)(x - m_logicalOriginX) * m_scaleX) * m_signX + m_deviceOriginX;
     }
     // Without device translation, for backing pixmap purposes
     wxCoord XLOG2DEV_2(wxCoord x) const
     {
-        wxCoord new_x = x - m_logicalOriginX;
-        if (new_x > 0)
-            return (wxCoord)((double)(new_x) * m_scaleX + 0.5) * m_signX;
-        else
-            return (wxCoord)((double)(new_x) * m_scaleX - 0.5) * m_signX;
+        return wxCoordRound((double)(x - m_logicalOriginX) * m_scaleX) * m_signX;
     }
     wxCoord XLOG2DEVREL(wxCoord x) const
     {
-        if (x > 0)
-            return (wxCoord)((double)(x) * m_scaleX + 0.5);
-        else
-            return (wxCoord)((double)(x) * m_scaleX - 0.5);
+        return wxCoordRound((double)(x) * m_scaleX);
     }
     wxCoord YLOG2DEV(wxCoord y) const
     {
-        wxCoord new_y = y - m_logicalOriginY;
-        if (new_y > 0)
-            return (wxCoord)((double)(new_y) * m_scaleY + 0.5) * m_signY + m_deviceOriginY;
-        else
-            return (wxCoord)((double)(new_y) * m_scaleY - 0.5) * m_signY + m_deviceOriginY;
+        return wxCoordRound((double)(y - m_logicalOriginY) * m_scaleY) * m_signY + m_deviceOriginY;
     }
     // Without device translation, for backing pixmap purposes
     wxCoord YLOG2DEV_2(wxCoord y) const
     {
-        wxCoord new_y = y - m_logicalOriginY;
-        if (new_y > 0)
-            return (wxCoord)((double)(new_y) * m_scaleY + 0.5) * m_signY;
-        else
-            return (wxCoord)((double)(new_y) * m_scaleY - 0.5) * m_signY;
+        return wxCoordRound((double)(y - m_logicalOriginY) * m_scaleY) * m_signY;
     }
     wxCoord YLOG2DEVREL(wxCoord y) const
     {
-        if (y > 0)
-            return (wxCoord)((double)(y) * m_scaleY + 0.5);
-        else
-            return (wxCoord)((double)(y) * m_scaleY - 0.5);
+        return wxCoordRound((double)(y) * m_scaleY);
     }
 
Index: include/wx/x11/dc.h
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/include/wx/x11/dc.h,v
retrieving revision 1.10
diff -b -u -2 -r1.10 dc.h
--- include/wx/x11/dc.h	2006/09/05 20:46:53	1.10
+++ include/wx/x11/dc.h	2006/10/29 01:45:09
@@ -66,61 +66,33 @@
     wxCoord XDEV2LOG(wxCoord x) const
     {
-        wxCoord new_x = x - m_deviceOriginX;
-        if (new_x > 0)
-            return (wxCoord)((double)(new_x) / m_scaleX + 0.5) * m_signX + m_logicalOriginX;
-        else
-            return (wxCoord)((double)(new_x) / m_scaleX - 0.5) * m_signX + m_logicalOriginX;
+        return wxCoordRound((double)(x - m_deviceOriginX) / m_scaleX) * m_signX + m_logicalOriginX;
     }
     wxCoord XDEV2LOGREL(wxCoord x) const
     {
-        if (x > 0)
-            return (wxCoord)((double)(x) / m_scaleX + 0.5);
-        else
-            return (wxCoord)((double)(x) / m_scaleX - 0.5);
+        return wxCoordRound((double)(x) / m_scaleX);
     }
     wxCoord YDEV2LOG(wxCoord y) const
     {
-        wxCoord new_y = y - m_deviceOriginY;
-        if (new_y > 0)
-            return (wxCoord)((double)(new_y) / m_scaleY + 0.5) * m_signY + m_logicalOriginY;
-        else
-            return (wxCoord)((double)(new_y) / m_scaleY - 0.5) * m_signY + m_logicalOriginY;
+        return wxCoordRound((double)(y - m_deviceOriginY) / m_scaleY) * m_signY + m_logicalOriginY;
     }
     wxCoord YDEV2LOGREL(wxCoord y) const
     {
-        if (y > 0)
-            return (wxCoord)((double)(y) / m_scaleY + 0.5);
-        else
-            return (wxCoord)((double)(y) / m_scaleY - 0.5);
+        return wCoordRound((double)(y) / m_scaleY);
     }
     wxCoord XLOG2DEV(wxCoord x) const
     {
-        wxCoord new_x = x - m_logicalOriginX;
-        if (new_x > 0)
-            return (wxCoord)((double)(new_x) * m_scaleX + 0.5) * m_signX + m_deviceOriginX;
-        else
-            return (wxCoord)((double)(new_x) * m_scaleX - 0.5) * m_signX + m_deviceOriginX;
+        return wxCoordRound((double)(x - m_logicalOriginX) * m_scaleX) * m_signX + m_deviceOriginX;
     }
     wxCoord XLOG2DEVREL(wxCoord x) const
     {
-        if (x > 0)
-            return (wxCoord)((double)(x) * m_scaleX + 0.5);
-        else
-            return (wxCoord)((double)(x) * m_scaleX - 0.5);
+        return wxCoordRound((double)(x) * m_scaleX);
     }
     wxCoord YLOG2DEV(wxCoord y) const
     {
-        wxCoord new_y = y - m_logicalOriginY;
-        if (new_y > 0)
-            return (wxCoord)((double)(new_y) * m_scaleY + 0.5) * m_signY + m_deviceOriginY;
-        else
-            return (wxCoord)((double)(new_y) * m_scaleY - 0.5) * m_signY + m_deviceOriginY;
+        return wxCoordRound((double)(y - m_logicalOriginY) * m_scaleY) * m_signY + m_deviceOriginY;
     }
     wxCoord YLOG2DEVREL(wxCoord y) const
     {
-        if (y > 0)
-            return (wxCoord)((double)(y) * m_scaleY + 0.5);
-        else
-            return (wxCoord)((double)(y) * m_scaleY - 0.5);
+        return wxCoordRound((double)(y) * m_scaleY);
     }
 
Index: src/common/prntbase.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/prntbase.cpp,v
retrieving revision 1.105
diff -b -u -2 -r1.105 prntbase.cpp
--- src/common/prntbase.cpp	2006/10/27 14:16:39	1.105
+++ src/common/prntbase.cpp	2006/10/29 01:45:10
@@ -618,8 +618,8 @@
     float mmToDeviceX = float(pw) / mw;
     float mmToDeviceY = float(ph) / mh;
-    wxRect pageMarginsRect(paperRect.x + wxCoord(mmToDeviceX * topLeft.x),
-        paperRect.y + wxCoord(mmToDeviceY * topLeft.y),
-        paperRect.width - wxCoord(mmToDeviceX * (topLeft.x + bottomRight.x)),
-        paperRect.height - wxCoord(mmToDeviceY * (topLeft.y + bottomRight.y)));
+    wxRect pageMarginsRect(paperRect.x + wxCoordRound(mmToDeviceX * topLeft.x),
+        paperRect.y + wxCoordRound(mmToDeviceY * topLeft.y),
+        paperRect.width - wxCoordRound(mmToDeviceX * (topLeft.x + bottomRight.x)),
+        paperRect.height - wxCoordRound(mmToDeviceY * (topLeft.y + bottomRight.y)));
     wxCoord w, h;
     m_printoutDC->GetSize(&w, &h);
@@ -709,8 +709,8 @@
     float scaleX = float(w) / pw;
     float scaleY = float(h) / ph;
-    return wxRect(m_printoutDC->DeviceToLogicalX(wxCoord(paperRect.x * scaleX)), 
-        m_printoutDC->DeviceToLogicalY(wxCoord(paperRect.y * scaleY)), 
-        m_printoutDC->DeviceToLogicalXRel(wxCoord(paperRect.width * scaleX)), 
-        m_printoutDC->DeviceToLogicalYRel(wxCoord(paperRect.height * scaleY)));
+    return wxRect(m_printoutDC->DeviceToLogicalX(wxCoordRound(paperRect.x * scaleX)), 
+        m_printoutDC->DeviceToLogicalY(wxCoordRound(paperRect.y * scaleY)), 
+        m_printoutDC->DeviceToLogicalXRel(wxCoordRound(paperRect.width * scaleX)), 
+        m_printoutDC->DeviceToLogicalYRel(wxCoordRound(paperRect.height * scaleY)));
 }
 
@@ -741,8 +741,8 @@
     float mmToDeviceX = float(pw) / mw;
     float mmToDeviceY = float(ph) / mh;
-    wxRect pageMarginsRect(paperRect.x + wxCoord(mmToDeviceX * topLeft.x),
-        paperRect.y + wxCoord(mmToDeviceY * topLeft.y),
-        paperRect.width - wxCoord(mmToDeviceX * (topLeft.x + bottomRight.x)),
-        paperRect.height - wxCoord(mmToDeviceY * (topLeft.y + bottomRight.y)));
+    wxRect pageMarginsRect(paperRect.x + wxCoordRound(mmToDeviceX * topLeft.x),
+        paperRect.y + wxCoordRound(mmToDeviceY * topLeft.y),
+        paperRect.width - wxCoordRound(mmToDeviceX * (topLeft.x + bottomRight.x)),
+        paperRect.height - wxCoordRound(mmToDeviceY * (topLeft.y + bottomRight.y)));
     wxCoord w, h;
     m_printoutDC->GetSize(&w, &h);
@@ -757,8 +757,8 @@
     float scaleX = float(w) / pw;
     float scaleY = float(h) / ph;
-    return wxRect(m_printoutDC->DeviceToLogicalX(wxCoord(pageMarginsRect.x * scaleX)), 
-        m_printoutDC->DeviceToLogicalY(wxCoord(pageMarginsRect.y * scaleY)), 
-        m_printoutDC->DeviceToLogicalXRel(wxCoord(pageMarginsRect.width * scaleX)), 
-        m_printoutDC->DeviceToLogicalYRel(wxCoord(pageMarginsRect.height * scaleY)));
+    return wxRect(m_printoutDC->DeviceToLogicalX(wxCoordRound(pageMarginsRect.x * scaleX)), 
+        m_printoutDC->DeviceToLogicalY(wxCoordRound(pageMarginsRect.y * scaleY)), 
+        m_printoutDC->DeviceToLogicalXRel(wxCoordRound(pageMarginsRect.width * scaleX)), 
+        m_printoutDC->DeviceToLogicalYRel(wxCoordRound(pageMarginsRect.height * scaleY)));
 }
 
Index: src/mac/carbon/dccg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/mac/carbon/dccg.cpp,v
retrieving revision 1.60
diff -b -u -2 -r1.60 dccg.cpp
--- src/mac/carbon/dccg.cpp	2006/10/20 11:07:00	1.60
+++ src/mac/carbon/dccg.cpp	2006/10/29 01:45:10
@@ -2287,68 +2287,40 @@
 wxCoord wxDCBase::DeviceToLogicalX(wxCoord x) const
 {
-        long new_x = x - m_deviceOriginX;
-        if (new_x > 0)
-            return (wxCoord)((double)new_x / m_scaleX + 0.5) * m_signX + m_logicalOriginX;
-        else
-            return (wxCoord)((double)new_x / m_scaleX - 0.5) * m_signX + m_logicalOriginX;
+    return wxCoordRound((double)(x - m_deviceOriginX) / m_scaleX) * m_signX + m_logicalOriginX;
 }
 
 wxCoord wxDCBase::DeviceToLogicalY(wxCoord y) const
 {
-	long new_y = y - m_deviceOriginY;
-	if (new_y > 0)
-		return (wxCoord)((double)new_y / m_scaleY + 0.5) * m_signY + m_logicalOriginY;
-	else
-		return (wxCoord)((double)new_y / m_scaleY - 0.5) * m_signY + m_logicalOriginY;
+    return wxCoordRound((double)(y - m_deviceOriginY) / m_scaleY) * m_signY + m_logicalOriginY;
 }
 
 wxCoord wxDCBase::DeviceToLogicalXRel(wxCoord x) const
 {
-	if (x > 0)
-		return (wxCoord)((double)x / m_scaleX + 0.5);
-	else
-		return (wxCoord)((double)x / m_scaleX - 0.5);
+    return wxCoordRound((double)(x) / m_scaleX);
 }
 
 wxCoord wxDCBase::DeviceToLogicalYRel(wxCoord y) const
 {
-	if (y > 0)
-		return (wxCoord)((double)y / m_scaleY + 0.5);
-	else
-		return (wxCoord)((double)y / m_scaleY - 0.5);
+    return wxCoordRound((double)(y) / m_scaleY);
 }
 
 wxCoord wxDCBase::LogicalToDeviceX(wxCoord x) const
 {
-	long new_x = x - m_logicalOriginX;
-	if (new_x > 0)
-		return (wxCoord)((double)new_x * m_scaleX + 0.5) * m_signX + m_deviceOriginX;
-	else
-		return (wxCoord)((double)new_x * m_scaleX - 0.5) * m_signX + m_deviceOriginX;
+    return wxCoordRound((double)(x - m_logicalOriginX) * m_scaleX) * m_signX + m_deviceOriginX;
 }
 
 wxCoord wxDCBase::LogicalToDeviceY(wxCoord y) const
 {
-	long new_y = y - m_logicalOriginY;
-	if (new_y > 0)
-		return (wxCoord)((double)new_y * m_scaleY + 0.5) * m_signY + m_deviceOriginY;
-	else
-		return (wxCoord)((double)new_y * m_scaleY - 0.5) * m_signY + m_deviceOriginY;
+    return wxCoordRound((double)(y - m_logicalOriginY) * m_scaleY) * m_signY + m_deviceOriginY;
 }
 
 wxCoord wxDCBase::LogicalToDeviceXRel(wxCoord x) const
 {
-	if (x > 0)
-		return (wxCoord)((double)x * m_scaleX + 0.5);
-	else
-		return (wxCoord)((double)x * m_scaleX - 0.5);
+    return wxCoordRound((double)(x) * m_scaleX);
 }
 
 wxCoord wxDCBase::LogicalToDeviceYRel(wxCoord y) const
 {
-        if (y > 0)
-            return (wxCoord)((double)y * m_scaleY + 0.5);
-        else
-            return (wxCoord)((double)y * m_scaleY - 0.5);
+    return wxCoordRound((double)(y) * m_scaleY);
 }
 

 	  	 
